api/utils/queryBuilder.ts

```diff
@@ -0,0 +1,34 @@
+import { PokemonQueryOptions } from "../interfaces/sort.interfaces";
+
+export function buildPokemonQuery({
+  sort,
+  page,
+  limit,
+  filters,
+  pokemonIds = [],
+}: PokemonQueryOptions) {
+  const skip = (page - 1) * limit;
+  const { name, user_id } = filters || {};
+
+  let query = `
+    SELECT p.*
+    FROM "Pokemon" p
+    WHERE 1=1
+  `;
+
+  if (user_id && pokemonIds.length > 0) {
+    query += ` AND p.id IN (${pokemonIds.map((pid) => `'${pid}'`).join(", ")})`;
+  }
+
+  if (name) {
+    query += ` AND p.name ILIKE '%${name}%'`;
+  }
+
+  if (sort) {
+    query += ` ORDER BY ${sort.field} ${sort.order.toUpperCase()}`;
+  }
+
+  query += ` LIMIT ${limit} OFFSET ${skip}`;
+
+  return query;
```

Comment:
i think you should build just the filters,
the LIMIT,OFFSET,ORDER BY are straightforward, the where is the complex part.
your trade off here is the clarity and the maintenance